components/system-broker/pkg/director/client.go

```diff
@@ -82,17 +83,15 @@ func (c *GraphQLClient) FetchApplication(ctx context.Context, id string) (*Appli
 		"auths",
 		"eventingConfiguration",
 	}))
-
 	app := ApplicationOutput{}
-
 	req := gcli.NewRequest(query)
 
-	err := c.gcli.Do(ctx, req, &app)
-	if err != nil {
+	if err := c.gcli.Do(ctx, req, &app); err != nil {
 		return nil, errors.Wrap(err, "while fetching application in gqlclient")
 	}
 	if app.Result == nil {
-		return nil, errors.New("failed to fetch application")
+		log.C(ctx).Warnf("Application with ID %s not found", id)
```

Comment:
probably its better to keep the client behavour as is - with the change the op will never be deleted